### ISSUE TITLE
Migrate to wagtail.snippets or wagtail-modeladmin 

### ISSUE THREAD

**Issue by UDAR-TomH** [jazzband/wagtailmenus#459]
Seeing as how wagtail.contrib.modeladmin is going away in Wagtail 6.0, are there plans to update this package to support either wagtail.snippets or move to wagtail-modeladmin? Would be interested to know the roadmap here.

**Comment by MrCordeiro** [comment]
We will likely transition to snippets as it's more aligned to the overall Wagtail 6 architecture.

wagtail-modeladmin feels more like a maintenance package and may not be updated in the future.

**Comment by dkirkham** [comment]
Is it perhaps worth a quick fix to make use of the decoupled `wagtail_modeladmin` package, as per the instructions at https://github.com/wagtail-nest/wagtail-modeladmin/blob/main/docs/installation.md?

This will at least remove the annoying deprecation messages while the more substantial changes are made to use snippets.

I'm happy to do this, and to assist (maybe with testing if you are making progress @frague59) with the snippet version.

**Comment by JanMalte** [comment]
> I'm working on it : https://github.com/frague59/wagtailmenus

Hi, are you still on it? Do you need any help?

**Comment by frague59** [comment]
... I'm on another project right now, I'm missed up in the templates used by the admin - I assume I'm a bit lazy on investigating on it.

And I've used the modeladmin external package for my project.

I've not gone so far !

**Comment by MrCordeiro** [comment]
I feel this migration won't be easy. Wagtailmenus is very coupled with wagtail-modeladmin and its base views. A few wheels will have to be reinvented for snippets.

Maybe we could split the problem into smaller issues and create a milestone for it.

**Comment by MrCordeiro** [comment]
As an end user, I prefer the 2nd option. I feel it's more explicit.

**Comment by dkirkham** [comment]
Despite my comment [above](https://github.com/jazzband/wagtailmenus/issues/459#issuecomment-1838470701), I am presently pursuing the first option.

Two reasons:

- in a single site deployment, it avoids the distracting intermediate index view which will only ever have one entry
- the PR I create should as much as possible retain existing behaviour, focusing on the ModelAdmin to Snippet change; we can look at UX and editor workflow changes as a separate issue and PR.

I'm just working through the template issues, trying to minimise or eliminate duplication of Wagtail templates that might create an ongoing maintenance burden. I'm also waiting on exactly what happens in https://github.com/wagtail/wagtail/pull/11270 to make sure the FlatMenu copy is consistent - I'll have to backport some code to support Wagtail 5.2 LTS, but hopefully that can be bypassed in Wagtail 6, making for a fairly short deprecation cycle (once 5.2 LTS drops from support).

**Comment by Krxplx** [comment]
I installed the third party library wagtail_modeladmin. The main menu is working well, but i cant edit flat menus in cms. Is there anyone with the same problem?

**Comment by dkirkham** [comment]
@vdboor, release 3.1.9 provides a transition to `wagtail_modeladmin` which works with Wagtail 5.1.

I have also prepared a PR to completely remove dependence on wagtail_modeladmin in Wagtail 5.2 LTS and 6.0. See #472. It is undergoing final testing and only needs a few documentation updates before it is ready.
